### PR TITLE
feat(docs): add docs.appendMarkdown for formatted writes without index math

### LINKED ISSUE #400: Feature request: docs.appendMarkdown — write formatted Docs content without manual index math
## Problem

Producing a formatted Google Doc today takes the two-step flow the `google-docs` skill teaches: `docs.writeText` with plain text, then `docs.formatText` with hand-computed 1-based character indices. The skill spends most of its length teaching the model index arithmetic ("count the \n characters..."), and any miscount styles the wrong range. For agents, markdown is the native output format — the index math is pure failure surface.

## Proposal

`docs.appendMarkdown(documentId, markdown, tabId?)`: converts a markdown string to natively formatted Docs content in a single call.

- Supported: headings (`#`–`######`), `**bold**`, `*italic*`, `~~strikethrough~~`, `` `inline code` ``, `[links](url)`, bullet (`- `) and numbered (`1. `) lists
- One `insertText` with the plain text, then one `batchUpdate` with `updateParagraphStyle` / `updateTextStyle` / `createParagraphBullets` requests computed from parse offsets — no client-side index math ever reaches the model
- `tabId` optional, default tab when omitted (same convention as `docs.writeText`)
- Not in scope for v1 (documented): tables, horizontal rules, images — they degrade gracefully to plain text / are skipped with counts in the response

No new OAuth scopes (`documents` already covers it). Complements rather than replaces `docs.formatText`, which remains the right tool for fine-grained edits to existing text.

I have a working implementation (pure parser module + service method) with Jest coverage ready to submit as a PR.

### PR DESCRIPTION
**Pull request by matheusrmorgado**
## Summary

Fixes #400. Adds `docs.appendMarkdown(documentId, markdown, tabId?)`: converts markdown into natively formatted Docs content in a single call.

```
docs.appendMarkdown({
  documentId: "...",
  markdown: "# Status\n\nShipping **on time**.\n\n- item one\n- item two"
})
```

Today the only way to produce formatted content is the two-step `docs.writeText` + `docs.formatText` flow, where the model hand-computes 1-based character indices (the `google-docs` skill spends most of its length teaching that arithmetic). For agents, markdown is the native output format — this removes the index math entirely.

## Design

- **Pure parser module** (`utils/markdown.ts`): blocks (headings 1-6, bullets, numbered, paragraphs, horizontal rules) with inline formats (`**bold**`, `*italic*`/`_italic_`, `~~strikethrough~~`, `` `code` ``, `[links](url)`) recorded as character ranges over the stripped text. Single-pass, non-nesting semantics, documented in the module.
- **Two batchUpdates total**: one `insertText` with the full stripped text, then one batch of `updateParagraphStyle` / `updateTextStyle` / `createParagraphBullets` requests computed from parse offsets. Consecutive list items of the same kind merge into single bullet ranges.
- **`tabId` optional** with the same convention/discovery as `docs.writeText` (flattened tab walk incl. child tabs; clear `Tab with ID x not found` error).
- **Graceful degradation, explicit in the tool description**: horizontal rules are skipped (no Docs API request exists for them) and counted in the response; tables are not parsed and fall through as plain paragraphs.
- **No new OAuth scopes** (`documents` already covers it). Registered in the `docs.write` feature group.

`docs.formatText` remains the right tool for fine-grained edits to *existing* text; this covers the much more common "write new formatted content" path.

## What's included

- `utils/markdown.ts` parser + `DocsService.appendMarkdown` following the existing service conventions (logging, `isError` shape, `TABS_FIELD_MASK` reuse)
- Tool registration with Zod schema
- 16 Jest tests: 11 parser tests (inline markers incl. unclosed/empty edge cases, links, heading levels, list items with leading whitespace, hr, pipe-row fall-through, mixed document) and 5 service tests (exact request shapes incl. computed ranges, tabId threading, tab-not-found, hr-only short-circuit, error path)
- `docs/index.md` entry

## Verification

```
npm run test && npm run lint && npm run format:check && npx tsc --noEmit --project workspace-server
```

All passing: 30 suites, 635 tests (619 existing + 16 new), eslint/prettier/tsc clean.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: workspace-server/src/services/DocsService.ts [hunk 1]
```diff
@@ -402,6 +403,244 @@ export class DocsService {
...
+      const insertAt = Math.max(1, (lastElement?.endIndex || 1) - 1);
+
+      // One insertText with the full plain text; offsets let us compute every
+      // style range afterwards.
+      let fullText = '';
+      const offsets: number[] = [];
+      for (const block of renderable) {
+        offsets.push(fullText.length);
+        fullText += block.text + '\n';
+      }
```
[the remark this answers is not in the document]
Comment (matheusrmorgado) [reply workspace-server/src/services/DocsService.ts]: Excellent catch — fixed in e9de78b. A newline is prepended when `insertAt > 1`, with offsets computed over the prefixed text so every style range shifts automatically. This prevents both the text merge and the block style (e.g. HEADING_1) leaking onto the existing last paragraph.

File: workspace-server/src/__tests__/services/DocsService.test.ts [hunk 2]
```diff
@@ -1158,4 +1158,149 @@ describe('DocsService', () => {
...
+
+      const insertCall = mockDocsAPI.documents.batchUpdate.mock.calls[0][0];
+      expect(insertCall.requestBody.requests[0].insertText.location).toEqual({
+        index: 9,
+        tabId: 't.target',
+      });
+      const styleCall = mockDocsAPI.documents.batchUpdate.mock.calls[1][0];
+      expect(
+        styleCall.requestBody.requests[0].updateParagraphStyle.range,
+      ).toEqual({ startIndex: 9, endIndex: 12, tabId: 't.target' });
```
[the remark this answers is not in the document]
Comment (matheusrmorgado) [reply workspace-server/src/__tests__/services/DocsService.test.ts]: Done in e9de78b: the tab test now asserts the prepended newline in the insertText payload and the shifted range (10..13).

File: workspace-server/src/utils/markdown.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      const m = LINK_RE.exec(input.slice(i));
+      if (m) {
+        const start = out.length;
+        out += m[1];
+        formats.push({ type: 'link', start, end: out.length, url: m[2] });
+        i += m[0].length;
+        continue;
+      }
+    }
+    if (tryMarker('**', 'bold')) continue;
```
[the remark this answers is not in the document]
Comment (matheusrmorgado) [reply workspace-server/src/utils/markdown.ts:81]: Done in e9de78b: `__bold__` supported (checked before the single `_` italic marker) with a parser test.

**Comment by bdoyle0182** [comment]
appendMarkdown as a dedicated individual tool seems odd. I would try to propose something more akin to formatting that is generic and markdown as a param to the tool is just one option for the model to format the input
